keps/sig-auth/4633-anonymous-auth-health-only/README.md

```diff
@@ -0,0 +1,920 @@
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+By default, requests to the `kube-apiserver` that are not rejected by
+configured authentication methods are treated as anonymous requests, and
+given a username of `system:anonymous` and a group of `system:unauthenticated`.
+
+This behavior is can be toggled on or off by using the `--anonymous-auth`
+boolean flag.
+
+We propose that kubernetes should allow users to configure which endpoints can
+be accessed anonymously and disable anonymous auth for all other endpoints.
```

Comment:
the bootstrap token mechanism (https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens) supported in core k8s relies on the interaction of anonymous clients with the kube-public/cluster-info configmap, that  holds JWT validation and a server validating CA.

just to confirm, with core k8s defaults in mind, what happens after the code from this KEP merges?
would anonymous clients still be able to access the cluster-info CM.

Reply:
Including the kubeadm scenario in use-cases would be nice to illustrate why configurable paths are useful

Reply:
Added to the user story section. Thanks!